js-exercises/object-utils/objectUtils.js

```diff
@@ -0,0 +1,104 @@
+//* Helper function
+function isObject(obj) {
+  return typeof obj === 'object' && !Array.isArray(obj) && obj !== null;
+}
+
+function isArrayOfObjects(arrayOfObjects) {
+  if (Array.isArray(arrayOfObjects)) {
+    // eslint-disable-next-line consistent-return
+    arrayOfObjects.forEach(ele => {
+      if (!isObject(ele)) {
+        const message = `Expected all array items to be objects, found at least one ${typeof ele}, instead`;
+        return { hasError: true, message };
+      }
+    });
+  } else {
+    const message = `Expected array of objects, recieved ${typeof arrayOfObjects}`;
+    return { hasError: true, message };
+  }
+
+  return { hasError: false, message: '' };
+}
+
+//* Actual implementations
+function map(obj, callback) {
+  if (!isObject(obj)) {
+    throw new TypeError(`Expected Object, recieved ${typeof obj}`);
+  }
+
+  const result = {};
+  for (const property of Object.entries(obj)) {
```

Comment:
Here we can iterate object properties direct with :
`for (const property of obj) `

If you have used `Object.entries(obj)` for a specific reason then it's ok otherwise it just extra effort here

Reply:
Yes, that is true.

The reason I've used `entries` is, it gives me both `key` and `value` pair. 

Also, `for-of` only works with arrays, I think 🤔

I could have used `for-in` but it's not reliable + it doesn't give `key` `value` pair but instead, just `key`.

Hope that makes sense.

Thank you for taking your time. Much appreciated!! 😌

Reply:
Yes understood, the `key,valye` pair reason makes sense, Also I wrote `of` instead of `in` sorry for the confusion  😅

Reply:
I was working on my solutions as well I had a few confusion but that got cleared with your code. I'll let you know for the review once it's done.

Reply:
Sure, thing!